room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacRoundEnv.kt

```diff
@@ -47,4 +52,51 @@ internal class JavacRoundEnv(
             env.wrapTypeElement(MoreElements.asType(it))
         }.toSet()
     }
+
+    override fun getTypeElementsAnnotatedWith(klass: KClass<out Annotation>): Set<XTypeElement> {
+        return getTypeElementsAnnotatedWith(
+            annotationQualifiedName = klass.java.name
+        )
+    }
+
+    override fun getElementsAnnotatedWith(klass: KClass<out Annotation>): Set<XElement> {
+        val elements = delegate.getElementsAnnotatedWith(klass.java)
+        return wrapElements(elements, annotationName = { klass.java.name })
+    }
+
+    override fun getElementsAnnotatedWith(annotationQualifiedName: String): Set<XElement> {
+        val element = env.elementUtils.getTypeElement(annotationQualifiedName)
+            ?: error("Cannot find TypeElement: $annotationQualifiedName")
+
+        val elements = delegate.getElementsAnnotatedWith(element)
+
+        return wrapElements(elements, annotationName = { annotationQualifiedName })
+    }
+
+    private inline fun wrapElements(
+        result: Set<Element>,
+        annotationName: () -> String
+    ): Set<XElement> {
+
+        return result.map { element ->
+            when (element) {
+                is VariableElement -> {
+                    env.wrapVariableElement(element)
+                }
+                is TypeElement -> {
+                    env.wrapTypeElement(element)
+                }
+                is ExecutableElement -> {
+                    env.wrapExecutableElement(element)
+                }
+                is PackageElement -> {
+                    error(
+                        "Cannot get elements with annotation ${annotationName()}. Package " +
+                            "elements are not supported by KSP."
```

Comment:
maybe instead of KSP, we can put XProcessing to avoid confusion? (since they'll hit this code path only with JavaAP)